cmd/hadolintExecute_test.go

```diff
@@ -0,0 +1,84 @@
+package cmd
+
+import (
+	"testing"
+
+	"github.com/SAP/jenkins-library/pkg/hadolint/mocks"
+	piperMocks "github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+func TestRunHadolintExecute(t *testing.T) {
+	t.Run("default", func(t *testing.T) {
+		// init
+		fileMock := &mocks.HadolintPiperFileUtils{}
+		clientMock := &mocks.HadolintClient{}
+		runnerMock := &piperMocks.ExecMockRunner{}
+		config := hadolintExecuteOptions{
+			DockerFile:        "./Dockerfile",   // default
+			ConfigurationFile: ".hadolint.yaml", // default
+		}
+
+		fileMock.
+			On("FileExists", config.ConfigurationFile).Return(false, nil)
+
+		// test
+		err := runHadolint(config, hadolintUtils{
+			HadolintPiperFileUtils: fileMock,
+			HadolintClient:         clientMock,
+			hadolintRunner:         runnerMock,
+		})
+		// assert
+		assert.NoError(t, err)
+		assert.NotEmpty(t, runnerMock.Calls)
+		assert.Equal(t, 1, len(runnerMock.Calls))
+		assert.Equal(t, "hadolint", runnerMock.Calls[0].Exec)
+		assert.Contains(t, runnerMock.Calls[0].Params, config.DockerFile)
+		assert.Contains(t, runnerMock.Calls[0].Params, "--format")
+		assert.Contains(t, runnerMock.Calls[0].Params, "checkstyle")
+		assert.NotContains(t, runnerMock.Calls[0].Params, "--config")
+		assert.NotContains(t, runnerMock.Calls[0].Params, config.ConfigurationFile)
+		fileMock.AssertExpectations(t)
+		clientMock.AssertExpectations(t)
```

Comment:
At least to me it isn't clear from the test case code what those expectations are.

Reply:
The function is commented as follows:
> AssertExpectations asserts that everything specified with On and Return was in fact called as expected. Calls may have occurred in any order.

Shall I add this as inline comment?